public/index.html

```diff
@@ -0,0 +1,82 @@
+<!-- public/index.html -->
+<!DOCTYPE html>
+<html lang="en">
+<head>
+<meta charset="UTF-8">
+<title>Chat App</title>
+<style>
+  body {
+    background-color: black;
+    color: white;
+    font-family: Arial, sans-serif;
+  }
+  #chat {
+    max-width: 600px;
+    margin: 20px auto;
+    padding: 10px;
+    border: 1px solid #333;
+  }
+  #messages {
+    height: 300px;
+    overflow-y: auto;
+    margin-bottom: 10px;
+    padding: 5px;
+    background: #111;
+  }
+  #messageInput {
+    width: calc(100% - 60px);
+    margin-right: 10px;
+    background-color: dimgray;
+    color:white;
+  }
+  button {
+    width: 50px;
+    background-color: dimgray;
+    color:white;
+  }
+</style>
+</head>
+<body>
+<h1> NodeJS Cortext sample app </h1>
+<div id="chat">
+  <div id="messages"></div>
+  <input type="text" id="messageInput">
+  <button onclick="sendMessage()">Send</button>
+</div>
+
+<script>
+let conversationHistory = [];
+
+function sendMessage() {
+  const messageInput = document.getElementById('messageInput');
+  const userMessage = messageInput.value;
+  messageInput.value = '';
+
+  // Add the user's message to the conversation history
+  conversationHistory.push({ role: 'user', content: userMessage });
+  console.log(conversationHistory);
+  // Display the user's message
+  const messagesDiv = document.getElementById('messages');
+  messagesDiv.innerHTML += `<div><b>User</b>: ${userMessage}</div>`;
+
+  // Send the conversation history to the server
+  fetch('/sendMessage', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({ messages: conversationHistory })
+  })
```

Comment:
**Performance Issue**: The POST request in the sendMessage function sends the entire conversation history to the server each time a new message is added. This approach increases the payload size with each message, leading to inefficient use of network resources and potential performance degradation as the conversation history grows. <br> **Fix**: Modify the POST request to only send the new message rather than the entire conversation history. This change will reduce the data sent over the network, improving the performance and scalability of the application. <br> **Code Suggestion**: 
 ```
 -    body: JSON.stringify({ messages: conversationHistory })
 +    body: JSON.stringify({ message: userMessage })
 ```